server/src/test/java/org/elasticsearch/index/mapper/NestedObjectMapperTests.java

```diff
@@ -1737,6 +1737,30 @@ public void testSyntheticNestedWithIncludeInRoot() throws IOException {
             {"path":{"bar":"B","foo":"A"}}""", syntheticSource);
     }
 
+    public void testSyntheticNestedWithEmptyObject() throws IOException {
+        DocumentMapper documentMapper = createMapperService(syntheticSourceMapping(b -> {
+            b.startObject("path").field("type", "nested");
+            {
+                b.startObject("properties");
+                {
+                    b.startObject("foo").field("type", "keyword").endObject();
+                }
+                b.endObject();
+            }
+            b.endObject();
+        })).documentMapper();
+        var syntheticSource = syntheticSource(documentMapper, b -> {
+            b.startArray("path");
+            {
+                b.startObject().field("foo", "A").endObject();
+                b.startObject().nullField("foo").endObject();
+            }
+            b.endArray();
+        });
+        assertEquals("""
+            {"path":[{"foo":"A"},{}]}""", syntheticSource);
+    }
```

Comment:
We should also test with a single child like:
```
 {
            var syntheticSource = syntheticSource(documentMapper, b -> {
                b.startArray("path");
                {
                    b.startObject().nullField("foo").endObject();
                }
                b.endArray();
            });
            assertEquals("""
                    {"path":{}}""", syntheticSource);
        }
```

Should we also have the same tests with object (instead of nested) mappers?

Reply:
We test both objects and nested with this configuration using our integration tests (that's how i found this).